designs/unice/src/back.js

```diff
@@ -1,4 +1,321 @@
+  
+  // back curve and/or back dip
+  // back curve and back dip accomplish the same goal in two different ways:
+  // goal: reduce back height at the center
+  // back dip: curve waist band (pattern part is smile-shaped)
+  // back curve: start by curving the waist band, then straigthten the part (top of pattern part is roughly horizontal)
+  // if combined: apply back curve first, then back dip
+  
+  let backDipOrCurve1, backDipOrCurve2
+  if (options.backCurve > 0) {
+    backDipOrCurve1 = options.backCurve
+    backDipOrCurve2 = options.backDip
+  } else {
+    backDipOrCurve1 = options.backDip
+    backDipOrCurve2 = 0
+  }
 
+  // derive side seam from front (*before* the dip was converted into a curve)
+  points.backWaistMid = new Point(measurements.seat / 4, 0)
+  points.backWaistBandLeft = new Point(
+    store.get('sideSeamWaist').x / options.backToFrontWidth,
+    store.get('sideSeamWaist').y
+  )
+  points.backLegOpeningLeft = new Point(
+    store.get('sideSeamHip').x / options.backToFrontWidth,
+    store.get('sideSeamHip').y
+  )
+  
+  points.backWaistBandRight = points.backWaistBandLeft.flipX(points.backWaistMid)
+
+  points.backWaistBandMid = points.backWaistBandLeft
+    .shiftFractionTowards(points.backWaistBandRight, 0.5)
+    .shift(270, measurements.waistToUpperLeg * backDipOrCurve1 * store.get('yScaleReduced')) /* Waist band dip (or curve)*/
+    
+  // unlike front exposure (i.e. taperToGusset), curvature of the leg opening should not depend on backCurve
+  // ==> apply curvature first, then calculate control points
+  // straighten the part (if needed)
+  let curveAsAngle
+  if (options.backCurve > 0) {
+    // create copies of the original points
+    points.backWaistBandLeftPreCurve = points.backWaistBandLeft
+    points.backLegOpeningLeftPreCurve = points.backLegOpeningLeft
+    
+    // convert the dip to a curve
+    curveAsAngle = points.backWaistBandLeft.angle(points.backWaistBandMid)
+    points.backWaistBandLeft = points.backWaistBandLeft.rotate(-curveAsAngle,points.backWaistBandMid)
+    points.backLegOpeningLeft = points.backLegOpeningLeft.rotate(-curveAsAngle,points.backWaistBandMid)
+
+    // (ignore these two TODOs for now)
+    // TODO: recalculate the back height and portions 'above' and 'below'
+    // TODO: adjust the height of the 'below' part
+  } else {
+    curveAsAngle = 0
+  }
+  // recalculate backWaistBandRight  
+  points.backWaistBandRight = points.backWaistBandLeft.flipX(points.backWaistMid)
+  
+
+  // if back is both curved and dipped, apply the dip
+  // backDipOrCurve2 is zero if this isn't needed, so apply regardless of value
+  points.backWaistBandMid = points.backWaistBandMid.shift(270, measurements.waistToUpperLeg * backDipOrCurve2 * store.get('yScaleReduced'))        
+  
+  // report the expected 'drop' in waist band from side to center (when worn) 
+  raise.info(['expectedDropFromWaistToCenterBack', units(measurements.waistToUpperLeg * (options.backCurve + options.backDip))])  
```

Comment:
This currently prints an events log entry like: `expectedDropFromWaistToCenterBack0.88cm`

If you wanted to make it look prettier, you could change it to something like:
```js
  const measurementString = units(measurements.waistToUpperLeg * 
    (options.backCurve + options.backDip))
  raise.info("The expected drop for the back center waist is " + measurementString + ".")
``` 

to produce `The expected drop for the back center waist is 0.88cm.`

Reply:
I thought I read somewhere that the strings are supposed to be defined elsewhere, to make translation easier... Am I mistaken?

Reply:
Correct, but we'll add that in v3.

That being said, it's a best practice to split translatable strings and units.

See: https://freesewing.dev/reference/api/hooks/inserttext

Reply:
Given that information, it is your choice to leave your code as is (to make it easier to have translated strings in the future) or to change it (to make the message text prettier now). Either way is fine.

Reply:
I'll keep it the way it is, then.